stl/inc/queue

```diff
@@ -318,6 +375,19 @@ public:
         _STD push_heap(c.begin(), c.end(), comp);
     }
 
+#ifdef __cpp_lib_containers_ranges
+    template <_Container_compatible_range<_Ty> _Rng>
+    void push_range(_Rng&& _Range) { // LWG issue submitted but unnumbered as of 2022-06-20
+        if constexpr (requires { c.append_range(_Range); }) {
+            c.append_range(_Range);
+        } else {
+            _RANGES copy(_Range, back_insert_iterator{c});
+        }
+
+        _STD make_heap(c.begin(), c.end(), comp);

```

Comment:
Is it suitable to optimize this when the count of appended elements is small enough? A sequence of calls to `push_heap` may be cheaper than `make_heap` in this case.

Reply:
That is the intent of the "as if by `make_heap`" wording I submitted for LWG-3723 (Note to self: correct the issue number above). When inserting `k` elements, it may be better to `push_heap` when `k log N < 3N`. That said, I'm not super excited to dedicate lots of time to this in this PR; I'll file an issue to followup with some perf investigation.

Reply:
This is #2826.